### PR TITLE
Product page checkout bug due to tier pricing and invalid session

### PR DESCRIPTION
**Pull request by johnmccombs1**

### CODE REVIEW

File: app/code/community/Bolt/Boltpay/Helper/Data.php [hunk 1]
```diff
@@ -144,7 +144,7 @@ public function buildOnCheckCallback($checkoutType, $isVirtualQuote = false)
      */
     public function buildOnSuccessCallback($successCustom, $checkoutType)
     {
-        $saveOrderUrl = $this->getMagentoUrl('boltpay/order/save');
+        $saveOrderUrl = $this->getMagentoUrl('boltpay/order/save' . ($checkoutType == Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_PRODUCT_PAGE ? '/product_page/true' : ''));
```
Comment (daisy1754) [inline app/code/community/Bolt/Boltpay/Helper/Data.php]: is `true` suffix necessary? It seems a bit redundant

File: app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml [hunk 2]
```diff
@@ -4,13 +4,27 @@
...
 var do_checks = 1;
-
+<?php $productTierPrice = $this->getProductTierPrice(); ?>
 var boltConfigPDP = {
     _jsonProductCart: <?php echo $this->getCartDataJsForProductPage(); ?>,
+    _tierPrices : <?php echo json_encode($productTierPrice) ?>,
+    getPrice : function (qty) {
+        qty = Number(qty);
+        var i = this._tierPrices.length;
+        while (i--)
```
Comment (daisy1754) [inline app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml]: can you add bracket in while here and if below?

File: app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml [hunk 3]
```diff
@@ -4,13 +4,29 @@
...
+
+        return this._jsonProductCart.items[0].price;
+    },
+
     increaseQty: function(number) {
         if (this._jsonProductCart) {
             this._jsonProductCart.items[0].quantity = parseInt(number);
-            this._jsonProductCart.total = parseFloat(this._jsonProductCart.total) * number;
+            if(this._tierPrices){
+                this._jsonProductCart.items[0].price = this.getPrice(number);
```
Comment (mccottry) [inline app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml]: @johnmccombs1 If you are getting a tier price, then why not name the method `getTierPrice`?

Comment (johnmccombs1) [reply app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml]: @mccottry this method is used to get the price (both regular and tier price) which is why I left it.

**Pull request merged**